### DISCUSSION TITLE
Capability of PyGEM

### DISCUSSION THREAD

**Discussion by BilalAnwar165** [mathLab/PyGeM#273] [General]
Hi, 

i am looking for good volume mesh deformers based on the morphed surface mesh. I have a mesh (let's suppose a wing) enclosed by a volume domain. The mesh is unstructured (Tri/tetra elements) with an inflation layer (quad/penta elements) near the boundary of wing to capture the viscous layer. I am dealing with a higher dimensional Shape Optimization problem. The optimizer will provide the updated  design variables of the wing in each iteration. The mesh deformer will update the surface mesh and the volume mesh. I want to ask Can pyGEM deal with such a mesh? The mesh is a OpenFoam Mesh.

Thanks in advance

**Comment by andreamola** [comment]
Hi Bilal,
I think in the past we have done what you want to do, using PyGeM.
Basically, what you need to do is placing a set of RBF source points on the
surface mesh you are deforming (which could be an stl triangulation or the
boundary mesh of OpenFOAM). You then use the RBF interpolation to propagate
the wall deformation inside the three dimensional domain. This has worked
for us in the past on OpenFOAM grids. Just make sure you play a bit with
the RBF parameters (the radius in particular) and to use an adequate, but
not excessive, amount of RBF source points on the surfaces.
Let me know if you have any other questions

             Andrea

**Reply by BilalAnwar165** [reply]
Thank you @andreamola for the reply. A couple of more questions I would like to ask, and it would be of great help if you can answer them, please.

1. Using PyGEM can I create a FFD box of arbitrary shape? or may be provide points of the box in Plot3D format? for example, I would like to create a wing fitted FFD box. The wing is tapered and swept wing.

2.  Is the PyGEM openfoam version specific? Can I use PyGEM with OpenFoam ESI version 2206? Also can it write the deformed mesh back to Openfoam format?

3. Can PyGEM handle the prism layers specifically?

P.S: I only want to use PyGEM as a backbox for mesh deformation!

I thank you for your help and cooperation!

Regards
Bilal
